aoc/pkg/advent/advent.go

```diff
@@ -0,0 +1,108 @@
+package advent
+
+import (
+	"aoc/pkg/structs"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+	"os"
+	"strings"
+)
+
+func buildClient(urlObj *url.URL) (http.Client, error) {
```

Comment:
Http client is thread safe, so you can use `*http.Client` and pass it anywhere you need by reference. It would avoid unnecessary copying the struct.

Reply:
hmm that is good to know. What's a good rule of thumb for when to return pointers vs a copy of the struct?

Reply:
I see a variety of advice on this. Here's how I do it.
Use copy unless...
- It's a struct that wouldn't change once declared and is passed around a lot. (Most clients, loggers, configs structs would fit this in my experience)
- The struct is so big that it causes degraded performance due to too much copying.